bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/AbstractComponentImpl.java

```diff
@@ -89,76 +92,6 @@ public String getExportedType() {
         return resource.getResourceType();
     }
 
-    /**
-     * Returns an auto generated component ID.
-     *
-     * The ID is the first 10 characters of an SHA-1 hexadecimal hash of the component path,
-     * prefixed with the component name. Example: title-810f3af321
-     *
-     * If the component is referenced, the path is taken to be a concatenation of the component path,
-     * with the path of the first parent context resource that exists on the page or in the template.
-     * This ensures the ID is unique if the same component is referenced multiple times on the same page or template.
-     *
-     * Collision
-     * ---------
-     * c = expected collisions
-     * c ~= (i^2)/(2m) - where i is the number of items and m is the number of possibilities for each item.
-     * m = 16^n - for a hexadecimal string, where n is the number of characters.
-     *
-     * For i = 1000 components with the same name, and n = 10 characters:
-     *
-     * c ~= (1000^2)/(2*(16^10))
-     * c ~= 0.00000045
-     *
-     * @return the auto generated component ID
-     */
-    private String generateId() {
```

Comment:
I would keep this method here as the ID is specific to a component, not just to compute the data for the data layer

Reply:
I think the functionality for generating the ID was moved so that It did not reside in the internal package such that it could be used by client projects.

If this remains internal to the core components then any custom components would need to safely recreate this logic.

Reply:
It makes sense to move this method to the new `ComponentUtils` class.